final_task/rss_reader/converter.py

```diff
@@ -0,0 +1,153 @@
+"""This module converts data to pdf"""
+
+from fpdf import FPDF
+import urllib.request as url
+import os
+import sys
+import datetime
+
+this_directory = os.path.abspath(os.path.dirname(__file__))
+font = 'ARIALUNI.TTF'
+
+win_font = r'C:\Windows\Fonts\arial.ttf'
+lin_font = r'/usr/share/fonts/dejavu/DejaVuSansCondensed.ttf'
+
+if sys.platform == 'win32':
+    if os.path.isfile(win_font):
+        fontpath = win_font
+    else:
+        fontpath = os.path.join(this_directory, font)
+
+if sys.platform == 'linux':
+    if os.path.isfile(lin_font):
+        fontpath = lin_font
+    else:
+        fontpath = os.path.join(this_directory, font)
+
+
+def convert_pdf(rss_news_clean: dict, path: str):
+    """This function creates pdf"""
+
+    pdf = FPDF()
+    pdf.add_font('arial_uni', '', fontpath, True)
+    pdf.set_margins(10, 10, 10)
+    counter = 1
+
+    for rss_news in rss_news_clean.values():
+        pdf.add_page(('P', 'A4'))
+
+        if rss_news['image']:
+            try:
+                url.urlretrieve(rss_news['image'], str(counter)+'imageTemp.jpg')
+                pdf.image(str(counter)+'imageTemp.jpg', 120, 12, 50, 50, link=rss_news['image'])
+                os.remove(str(counter)+'imageTemp.jpg')
+                counter += 1
+            except RuntimeError:
+                pdf.set_line_width(1)
+                pdf.rect(120, 12, 50, 50)
+                pdf.set_xy(120, 12)
+                pdf.set_font("Arial", "B", size=14)
+                pdf.multi_cell(w=50, h=12, align='C', txt=' \n IMAGE IS CORRUPTED')
+            except url.HTTPError:
+                pdf.set_line_width(1)
+                pdf.rect(120, 12, 50, 50)
+                pdf.set_xy(120, 12)
+                pdf.set_font("Arial", "B", size=14)
+                pdf.multi_cell(w=50, h=12, align='C', txt=' \n IMAGE IS CORRUPTED')
+            except url.URLError:
+                pdf.set_line_width(1)
+                pdf.rect(120, 12, 50, 50)
+                pdf.set_xy(120, 12)
+                pdf.set_font("Arial", "B", size=14)
+                pdf.multi_cell(w=50, h=12, align='C', txt=' \n NO INTERNET CONNECTION')
+            except ValueError:
+                pdf.set_line_width(1)
+                pdf.rect(120, 12, 50, 50)
+                pdf.set_xy(120, 12)
+                pdf.set_font("Arial", "B", size=14)
+                pdf.multi_cell(w=50, h=12, align='C', txt=' \n NO PICTURE AVAILABLE')
+
+        else:
+            pdf.set_line_width(1)
+            pdf.rect(120, 12, 50, 50)
+            pdf.set_xy(120, 12)
+            pdf.set_font("Arial", "B", size=14)
+            pdf.multi_cell(w=50, h=12, align='C', txt=' \n NO PICTURE AVAILABLE')
```

Comment:
There code inside each block is almost completely the same. It is better to move it to a separate function

Reply:
The was created and implemented, it will be updated during the next commit.